pulp_ansible/app/tasks/collections.py

```diff
@@ -118,41 +118,41 @@ def import_collection(
     """
     CollectionImport.objects.get_or_create(task_id=get_current_job().id)
 
-    artifact = Artifact.objects.get(pk=artifact_pk)
+    temp_file = PulpTemporaryFile.objects.get(pk=temp_file_pk)
     filename = CollectionFilename(expected_namespace, expected_name, expected_version)
-    log.info(f"Processing collection from {artifact.file.name}")
+    log.info(f"Processing collection from {temp_file.file.name}")
     user_facing_logger = logging.getLogger("pulp_ansible.app.tasks.collection.import_collection")
 
-    with _artifact_guard(artifact):
-        try:
-            with artifact.file.open() as artifact_file:
-                importer_result = process_collection(
-                    artifact_file, filename=filename, logger=user_facing_logger
-                )
+    try:
+        with temp_file.file.open() as artifact_file:
+            importer_result = process_collection(
+                artifact_file, filename=filename, logger=user_facing_logger
+            )
+            artifact = temp_file.to_artifact()
+            importer_result["artifact_url"] = reverse("artifacts-detail", args=[artifact.pk])
+            collection_version = create_collection_from_importer(importer_result)
 
-                importer_result["artifact_url"] = reverse("artifacts-detail", args=[artifact_pk])
-                collection_version = create_collection_from_importer(importer_result)
+    except ImporterError as exc:
+        log.info(f"Collection processing was not successfull: {exc}")
```

Comment:
I wonder if we should delete temp_file if there was an import error?

Reply:
oh yeah, I forgot it, thanks!